### PR TITLE
[POC] move json-builder to our repo

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | yes/no
| Deprecations? | yes/no
| Fixed tickets | #8 
| License       | MIT

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,11 +14,13 @@
     ],
     "require": {
         "php": "^5.6 || ^7.0",
-        "egeloen/json-builder": "^2.0 || ^3.0",
         "symfony/dependency-injection": "^2.0 || ^3.0 || ^4.0",
         "symfony/expression-language": "^2.7 || ^3.0 || ^4.0",
         "symfony/form": "^2.7 || ^3.0 || ^4.0",
-        "symfony/framework-bundle": "^2.7 || ^3.0 || ^4.0"
+        "symfony/framework-bundle": "^2.7 || ^3.0 || ^4.0",
+        "symfony/http-foundation": "^2.8 || ^3.4 || ^4.0",
```
Comment (ElectricMaxxx) [inline composer.json]: my IDE cries as i have deps, that are not requiered directly (routing, http-foundation). PropertyAccess is a requirement used by json-builder directly

Comment (kunicmarko20) [reply composer.json]: That should be fixed in a separated PR

**Comment by ElectricMaxxx** [comment]
Ohh they had been there? Ohh i did not seebthem. :-(

**Comment by kunicmarko20** [comment]
yes :smile: , just fix flintCI and let's merge it.

File: docs/usage/json-builder.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+The Ivory JSON builder allows you to build your JSON through the Symfony2 `PropertyAccess Component`_ while keeping the
+control of the value escaping.
+
+Create a builder
+~~~~~~~~~~~~~~~~
+
+To build some JSON, you will need to instanciate a builder:
+
+.. code-block:: php
+    use Ivory\JsonBuilder\JsonBuilder;
```
Comment (ElectricMaxxx) [inline docs/usage/json-builder.rst]: here.

File: docs/usage/json-builder.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+.. code-block:: php
+    $builder->reset();
+
+Example
+~~~~~~~
+
+If you still don't understand, a concrete example :)
+
+.. code-block:: php
+    use Ivory\JsonBuilder\JsonBuilder;
```
Comment (ElectricMaxxx) [inline docs/usage/json-builder.rst]: here

File: src/Resources/config/renderer.xml [hunk 4]
```diff
@@ -14,7 +14,7 @@
             <argument type="expression">container.hasParameter('locale') ? parameter('locale') : null</argument>
         </service>
 
-        <service id="fos_ck_editor.renderer.json_builder" class="FOS\JsonBuilder\JsonBuilder">
```
Comment (ElectricMaxxx) [inline src/Resources/config/renderer.xml]: here.

File: tests/Renderer/CKEditorRendererTest.php [hunk 5]
```diff
@@ -104,14 +104,14 @@ public function testLegacyContstructor()
         $container->expects($this->exactly(5))
             ->method('get')
             ->withConsecutive(
-                ['fos_ck_editor.renderer.json_builder'],
+                ['ivory_ck_editor.renderer.json_builder'],
```
Comment (ElectricMaxxx) [inline tests/Renderer/CKEditorRendererTest.php]: here

File: tests/Renderer/CKEditorRendererTest.php [hunk 6]
```diff
@@ -104,14 +104,14 @@ public function testLegacyContstructor()
...
                 ['router'],
                 ['assets.packages'],
                 ['request_stack'],
                 ['templating']
             )
             ->willReturnMap([
-                ['fos_ck_editor.renderer.json_builder', new JsonBuilder()],
```
Comment (ElectricMaxxx) [inline tests/Renderer/CKEditorRendererTest.php]: here.

File: src/Resources/config/renderer.xml [hunk 7]
```diff
@@ -14,7 +14,7 @@
...
+        <service id="fox_ck_editor.renderer.json_builder" class="FOS\CKEditorBundle\Builder\JsonBuilder">
```
Comment (kunicmarko20) [inline src/Resources/config/renderer.xml]: `fos`*

**Comment by ElectricMaxxx** [comment]
No i am back on the good way, had one change with one replacement to much. now it will be green later.

**Comment by ElectricMaxxx** [comment]
I will merge as travis comes green, lets fix scrutinizer somewhere else

**Pull request merged**